ingester/ingester.go

```diff
@@ -506,11 +506,15 @@ func (i *Ingester) flushUser(userID string, immediate bool) {
 }
 
 func (i *Ingester) flushSeries(u *userState, fp model.Fingerprint, series *memorySeries, immediate bool) {
-
 	// Enqueue this series flushing if the oldest chunk is older than the threshold
 
 	u.fpLocker.Lock(fp)
-	firstTime := series.head().FirstTime()
+	if len(series.chunkDescs) <= 0 {
+		u.fpLocker.Unlock(fp)
```

Comment:
It sucks that there now have to be two distinct calls to `Unlock`, but I can't think of a better way right now.

(Well, you could define a method on `memorySeries` that returns `(firstTime, ok)`, then define `flush` as `ok && immediate || model.Now().Sub(firstTime) > i.cfg.MaxChunkAge`, but maybe a bit much.)